arc-mlir/src/tests/ifs.mlir

```diff
@@ -0,0 +1,16 @@
+// RUN: arc-mlir %s -split-input-file -verify-diagnostics
+
+module @toplevel {
+  func @main() {
+    %a = constant 0 : i1
+    %b = constant 3.14 : f64
+    %c = constant 0.693 : f64
+
+    "arc.if"(%a) ( {
+      "arc.block.result"(%b) : (f64) -> f64
+    },  {
+      "arc.block.result"(%c) : (f64) -> f64
+    }) : (i1) -> f64
```

Comment:
one question, can you have nested ifs?

Reply:
They work fine, I added a test for that.